NavigationSystem/main_ASPire.cpp

```diff
@@ -0,0 +1,297 @@
+#include <string>
+#include "SystemServices/Logger.h"
+#include "MessageBus/MessageBus.h"
+
+#if SIMULATION == 1
+ #include "Simulation/SimulationNode.h"
+#else
+ #include "Hardwares/HMC6343Node.h"
+ #include "Hardwares/GPSDNode.h"
+ #include "Hardwares/ActuatorNodeASPire.h"
+ #include "Hardwares/CAN_Services/CANService.h"
+ #include "Hardwares/CANWindsensorNode.h"
+#endif
+
+#include "Navigation/WaypointMgrNode.h"
+#include "WorldState/StateEstimationNode.h"
+#include "WorldState/WindStateNode.h"
+#include "HTTPSync/HTTPSyncNode.h"
+#include "Xbee/XbeeSyncNode.h"
+#include "WorldState/VesselStateNode.h"
+
+#if LOCAL_NAVIGATION_MODULE == 1
+  #include "Navigation/LocalNavigationModule/LocalNavigationModule.h"
+  #include "Navigation/LocalNavigationModule/Voters/WaypointVoter.h"
+  #include "Navigation/LocalNavigationModule/Voters/WindVoter.h"
+  #include "Navigation/LocalNavigationModule/Voters/ChannelVoter.h"
+  #include "Navigation/LocalNavigationModule/Voters/ProximityVoter.h"
+  #include "Navigation/LocalNavigationModule/Voters/MidRangeVoter.h"
+#else
+  #include "Navigation/LineFollowNode.h"
+#endif
+#include "LowLevelControllers/LowLevelControllerNodeASPire.h"
+
+
+#if USE_OPENCV_COLOR_DETECTION == 1
+#include "WorldState/obstacledetection/colorDetectionNode.h"
+#endif
+
+#include "Messages/DataRequestMsg.h"
+#include "DataBase/DBHandler.h"
+#include "DataBase/DBLoggerNode.h"
+#include "Hardwares/MaestroController/MaestroController.h"
+#include "Xbee/Xbee.h"
+
+#define DISABLE_LOGGING 0
+
+enum class NodeImportance {
+	CRITICAL,
+	NOT_CRITICAL
+};
+
+///----------------------------------------------------------------------------------
+/// Initialises a node and shutsdown the program if a critical node fails.
+///
+/// @param node 			A pointer to the node to initialise
+/// @param nodeName 		A string name of the node, for logging purposes.
+/// @param importance 		Whether the node is a critcal node or not critical. If a
+///							critical node fails to initialise the program will
+///							shutdown.
+///
+///----------------------------------------------------------------------------------
+void initialiseNode(Node& node, const char* nodeName, NodeImportance importance)
+{
+	if(node.init())
+	{
+		Logger::info("Node: %s - init\t[OK]", nodeName);
+	}
+	else
+	{
+		Logger::error("Node: %s - init\t\t[FAILED]", nodeName);
+
+		if(importance == NodeImportance::CRITICAL)
+		{
+			Logger::error("Critical node failed to initialise, shutting down");
+			Logger::shutdown();
+			exit(1);
+		}
+	}
+}
+
+///----------------------------------------------------------------------------------
+/// Used for development of the Local Navigation Module
+///
+///----------------------------------------------------------------------------------
+#if LOCAL_NAVIGATION_MODULE==1
+void development_LocalNavigationModule( MessageBus& messageBus, DBHandler& dbHandler)
```

Comment:
Why a function localNavigationModule ?
I can understand it if there are only codes conserning the LocalNavigationModule but there are also codes that are used in both lnm and line_follow.